### PR TITLE
Fix Keycloak authentication management indexing

### PR DESCRIPTION
**Pull request by Skrekulko**
##### SUMMARY
This PR fixes an issue with indexation during the update of an authentication flow, more specifically a step (execution or sub-flow). The older behavior caused the to-be-updated step to change its index, even though the user didn't specify it.

Next (even though this is a refac), the provider ID according to Keycloak can only be **basic-flow** or **client-flow**. So I introduced choices for it.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
keycloak_authentication

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)?

The `choices` addition would be an entry in `minor_changes`, the bugfix an entry in `bugfixes`. Please note that if you keep all changes in one PR, the bugfix cannot be backported to community.general 6.x.y.

**Comment by Skrekulko** [comment]
I'll open a new PR for the minor changes, so the bugfix in this one can be backported.

File: changelogs/fragments/6734-keycloak-auth-management-indexing.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - Fix Keycloak authentication flow (step or sub-flow) indexing during update, if not specified by the user (https://github.com/ansible-collections/community.general/pull/6734).
```
Comment (felixfontein) [inline changelogs/fragments/6734-keycloak-auth-management-indexing.yml:2]:
```suggestion
  - keycloak_authentication - fix Keycloak authentication flow (step or sub-flow) indexing during update, if not specified by the user (https://github.com/ansible-collections/community.general/pull/6734).
```

**Review by felixfontein — COMMENTED** [review]
If nobody objects, I'll merge this in ~two weeks.

**Comment by felixfontein** [comment]
@Skrekulko thanks for your contribution!

**Pull request merged**